### PR TITLE
add c-styled header file to curve-sdk

### PR DESCRIPTION
**Pull request by xu-chaojie**
### What problem does this PR solve?

Issue Number: #1396 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by xu-chaojie** [comment]
recheck

File: include/client/libcurve_define.h [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/*
+ *  Copyright (c) 2020 NetEase Inc.
```
Comment (SeanHai) [inline include/client/libcurve_define.h:2]: 2022

File: include/client/libcurve_define.h [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+ */
+
+#ifndef INCLUDE_CLIENT_LIBCURVE_DEFINE_H_
+#define INCLUDE_CLIENT_LIBCURVE_DEFINE_H_
+
+#include <unistd.h>
+#include <stdint.h>
+
+enum LIBCURVE_ERROR {
+    // 操作成功
```
Comment (SeanHai) [inline include/client/libcurve_define.h:31]: Change to English by the way

**Review by wu-hanqing — APPROVED** [review]

File: include/client/libcurve_define.h (hunk 1 above)
Comment (xu-chaojie) [reply include/client/libcurve_define.h:2]: done

File: include/client/libcurve_define.h (hunk 2 above)
Comment (xu-chaojie) [reply include/client/libcurve_define.h:31]: done

**Review by SeanHai — APPROVED** [review]

**Pull request merged**